lib/stdlib/doc/src/lists.xml

```diff
@@ -1126,6 +1126,34 @@ splitwith(Pred, List) ->
 [[a,x,1],[b,y,2],[c,z,3]]</pre>
       </desc>
     </func>
+
+    <func>
+      <name name="groupby" arity="2" since=""/>
+      <fsummary>Splits the list into groups using a function as discriminator.</fsummary>
+      <desc>
+        <p>The result is a map where each key is given by <anno>Fun</anno> and each value is a list of elements. The order of elements within each list is preserved from the list.</p>
+        <p><em>Examples:</em></p>
+        <pre>
+> <input>lists:groupby(fun(X) -> X rem 2 end, [1,2,3]).</input>
+#{0 => [2], 1 => [1, 3]}
+> <input>lists:groupby(fun erlang:length/1, ["ant", "buffalo", "cat", "dingo"]).</input>
+#{3 => ["ant", "cat"], 5 => ["dingo"], 7 => ["buffalo"]}</pre>
+      </desc>
+    </func>
+
+    <func>
+      <name name="groupby" arity="3" since=""/>
+      <fsummary>Splits the list into groups using a function as discriminator.</fsummary>
+      <desc>
+        <p>The result is a map where each key is given by <anno>Fun</anno> and each value is a list of elements given by the <anno>ValueFun</anno>. The order of elements within each list is preserved from the list.</p>
+        <p><em>Examples:</em></p>
+        <pre>
+> <input>lists:groupby(fun(X) -> X rem 2 end, fun(X) -> X*X end, [1,2,3]).</input>
+#{0 => [4], 1 => [1, 9]}
+> <input>lists:groupby(fun erlang:length/1, fun([H | _]) -> H end, ["ant", "buffalo", "cat", "dingo"]).</input>
```

Comment:
use of `fun lists:reverse/1` as 2nd argument may make this example a bit more clear I think

Reply:
lists:groupby(fun erlang:length/1, fun reverse/1, ["ant", "buffalo", "cat", "dingo"])?

Reply:
yes. `fun lists:reverse/1`. Just IMO.

Reply:
Also, your current example actually does not produce the output
```
#{3 => ["a", "c"], 5 => ["d"], 7 => ["b"]}
```
It would produce lists of characters, not lists of strings
```
#{3 => [$a, $c], 5 => [$d], 7 => [$b]}
```
unless you change it the way you did in the test or wrap `H` in a list: `fun([H | _]) -> [H] end`